driver/groupnorm_driver.hpp

```diff
@@ -0,0 +1,424 @@
+#include <algorithm>
+#include <cstdlib>
+#include <cfloat>
+#include <memory>
+#include <miopen/tensor.hpp>
+#include <numeric>
+#include <vector>
+#include <../test/tensor_holder.hpp>
+#include "random.hpp"
+
+template <typename Tgpu, typename Tref>
+class GroupNormDriver : public Driver
+{
+public:
+    GroupNormDriver() : Driver()
+    {
+        miopenCreateTensorDescriptor(&inputDesc);
+        miopenCreateTensorDescriptor(&weightDesc);
+        miopenCreateTensorDescriptor(&biasDesc);
+        miopenCreateTensorDescriptor(&outputDesc);
+        miopenCreateTensorDescriptor(&meanDesc);
+        miopenCreateTensorDescriptor(&rstdDesc);
+
+        data_type = miopen_type<Tgpu>{};
+    }
+
+    int AddCmdLineArgs() override;
+    int ParseCmdLineArgs(int argc, char* argv[]) override;
+    InputFlags& GetInputFlags() override { return inflags; }
+
+    int GetandSetData() override;
+    std::vector<int> GetInputTensorLengthsFromCmdLine();
+
+    int AllocateBuffersAndCopy() override;
+
+    int RunForwardGPU() override;
+    int RunForwardCPU();
+
+    int RunBackwardGPU() override;
+
+    Tref GetTolerance();
+    int VerifyBackward() override;
+    int VerifyForward() override;
+    ~GroupNormDriver() override
+    {
+
+        miopenDestroyTensorDescriptor(inputDesc);
+        miopenDestroyTensorDescriptor(weightDesc);
+        miopenDestroyTensorDescriptor(biasDesc);
+        miopenDestroyTensorDescriptor(outputDesc);
+        miopenDestroyTensorDescriptor(meanDesc);
+        miopenDestroyTensorDescriptor(rstdDesc);
+    }
+
+private:
+    InputFlags inflags;
+
+    int forw;
+    int dim_size;
+
+    miopenTensorDescriptor_t inputDesc;
+    miopenTensorDescriptor_t weightDesc;
+    miopenTensorDescriptor_t biasDesc;
+    miopenTensorDescriptor_t outputDesc;
+    miopenTensorDescriptor_t meanDesc;
+    miopenTensorDescriptor_t rstdDesc;
+
+    std::unique_ptr<GPUMem> in_dev;
+    std::unique_ptr<GPUMem> weight_dev;
+    std::unique_ptr<GPUMem> bias_dev;
+    std::unique_ptr<GPUMem> out_dev;
+    std::unique_ptr<GPUMem> mean_dev;
+    std::unique_ptr<GPUMem> rstd_dev;
+
+    std::vector<Tgpu> in;
+    std::vector<Tgpu> weight;
+    std::vector<Tgpu> bias;
+    std::vector<Tgpu> out;
+    std::vector<Tref> mean;
+    std::vector<Tref> rstd;
+    std::vector<Tref> outhost;
+    std::vector<Tref> meanhost;
+    std::vector<Tref> rstdhost;
+
+    int num_groups;
+    float eps;
+    miopenNormMode_t mode;
+};
+
+template <typename Tgpu, typename Tref>
+int GroupNormDriver<Tgpu, Tref>::ParseCmdLineArgs(int argc, char* argv[])
+{
+    inflags.Parse(argc, argv);
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        miopenEnableProfiling(GetHandle(), true);
+    }
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int GroupNormDriver<Tgpu, Tref>::GetandSetData()
+{
+    num_groups = static_cast<int>(inflags.GetValueDouble("num_groups"));
+    eps        = static_cast<double>(inflags.GetValueDouble("eps"));
```

Comment:
`GetValueDouble` returns `double`, there is no need to extra cast to `double`.
But `eps` is float, so it has to be
```suggestion
    eps        = static_cast<float>(inflags.GetValueDouble("eps"));
```

Reply:
Fixed as per your comment.